Black Clovedle/ViewController.swift

```diff
@@ -6,14 +6,710 @@
 //
 
 import UIKit
+import Foundation
+import SwiftUI
+import CryptoKit
 
-class ViewController: UIViewController {
-
+class ViewController: UIViewController, UITableViewDelegate, UITextFieldDelegate {
+    
+    @IBOutlet weak var selectedItemLabel: UILabel!
+    @IBOutlet weak var guessingTableView: UITableView!
+    @IBOutlet weak var tryAgain: UILabel!
+    @IBOutlet weak var tableView: UITableView!
+    @IBOutlet weak var guessText: UITextField!
+    @IBOutlet weak var chevron: UIButton!
+    @IBOutlet weak var submit: UIButton!
+    var tapDismissView: UIView!
+    var correctName: String?
+    var correctGender: String?
+    var correctAffiliation: String?
+    var correctMagicAttribute: String?
+    var correctDebutArch: String?
+    var correctSpirit: String?
+    var correctImage: UIImage?
+    var numberOfGuesses = 0
+    var guesses = ["Asta"]
+    var gender = [UIImage]()
+    var affiliation = [UIImage]()
+    var magicAttribute = [UIImage]()
+    var debutArch = [UIImage]()
+    var spirit = [UIImage]()
+    var genderTextArray = [String]()
+    var affiliationTextArray = [String]()
+    var magicTextArray = [String]()
+    var spiritTextArray = [String]()
+    var debutArchTextArray = [String]()
+    var characterImagesArray = [UIImage]()
+    var randomCharacter = Character.self
+    let characterNames = ["Asta", "Yuno", "Noelle Silva", "Yami Sukehiro", "Mimosa Vermillion", "Luck Voltia", "Fuegoleon Vermillion", "Nozel Silva", "Charlotte Roselei", "William Vangeance", "Julius Novachrono", "Magna Swing", "Vanessa Enoteca", "Finral Roulacase", "Gauche Adlai", "Charmy Pappitson", "Gordon Agrippa", "Grey", "Secre Swallowtail", "Klaus Lunettes", "Hamon Caseus", "Alecdora Sandler", "Letoile Becquerel", "Langris Vaude", "Rhya The Disloyal", "Licht", "Vetto The Despair", "Fana The Hatred", "Sally", "Rades Spirito", "Valtos", "Rill Boismortier", "Kirsch Vermillion", "Zora Ideale", "Dorothy Unsworth", "Leopold Vermillion", "Damnatio Kira", "Dante Zogratis", "Liebe", "Zagred", "Gadjah", "Gueldre", "Henry", "Jack The Ripper", "Kaiser Granvorka", "Lily Aquaria", "Loropechika", "Mars", "Mereoleona Vermillion", "Nacht Faust", "Nebra Silva", "Ralph Niaflem", "Sekke Bronzazza", "Sol Marron", "Solid Silva", "Undine", "Vanica", "Zenon Zogratis"]
```

Comment:
A common convention to bloating your view controller is to have a class defined called `Constants` and define `characterNames` as a static property. Then you can refer to it anywhere by `Constants.characterNames`:

```
enum Constants { 
  static let characterNames = [...]
}
```

With that said, you have a whole list of characters defined in `CharacterInfo`, so I would find a way to delete all this stuff and, if you need the list of character names, define it as a computed variable in `CharacterInfo`. That way, if BlackClover comes out with a new season with a bunch of new characters, you'll only need to add it to the list in `CharacterInfo` and the computed variable will automatically include it. 

Ultimately, you want to have a single source of truth for the list of characters.